include/nbl/video/utilities/IUtilities.h

```diff
@@ -64,8 +64,17 @@ class IUtilities : public core::IReferenceCounted
         //! Remember to ensure a memory dependency between the command recorded here and any users (so fence wait, semaphore when submitting, pipeline barrier or event)
         inline core::smart_refctd_ptr<IGPUImage> createFilledDeviceLocalGPUImageOnDedMem(IGPUCommandBuffer* cmdbuf, IGPUImage::SCreationParams&& params, const IGPUBuffer* srcBuffer, uint32_t regionCount, const IGPUImage::SBufferCopy* pRegions)
         {
-            // Todo(achal): 
-            // dstImage's format should support VK_FORMAT_FEATURE_TRANSFER_DST_BIT
+            // This API check is temporary (or not?) since getFormatProperties is not
+            // yet implemented on OpenGL
+            if (srcBuffer->getAPIType() == EAT_VULKAN)
+            {
+                const auto reqFormatFeature = asset::EFF_TRANSFER_DST_BIT;
+                const auto& formatProps = srcBuffer->getOriginDevice()->getPhysicalDevice()->getFormatProperties(params.format);
```

Comment:
IUtilities has a `m_logicalDevice` which you can use instead of `getOriginDevice()`

Reply:
Resolved